api/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -51,6 +51,11 @@ type OpenTelemetryCollectorSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 
+	// SecurityContext will be set as the container security context
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	SecurityContext v1.SecurityContext `json:"securityContext,omitempty"`
```

Comment:
Sorry for catching this only now, but this should be a pointer. Otherwise, we won't know when it's been specified explicitly by the user. We need this signal if we ever decide to change the default value we pass down to the containers.

Reply:
Updated